src/Microsoft.AspNet.Mvc.TagHelpers/AnchorTagHelper.cs

```diff
@@ -102,15 +108,16 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                             ProtocolAttributeName,
                             HostAttributeName,
                             FragmentAttributeName,
-                            RouteAttributePrefix,
+                            RouteValuesPrefix,
                             Href));
                 }
             }
             else
             {
-                TagBuilder tagBuilder;
-                var routeValues = GetRouteValues(output, routePrefixedAttributes);
+                // Convert from Dictionary<string, string> to Dictionary<string, object>.
+                var routeValues = RouteValues.ToDictionary(kvp => kvp.Key, kvp => (object)kvp.Value);
```

Comment:
Likely irrelevant because `UrlHelper` creates its own dictionary but I'll add `StringComparer.OrdinalIgnoreCase` here and in the `<form/>` tag helper.